src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -65,6 +65,14 @@ public LanguageServer(SendToClient sendToClient, IPowerFxScopeFactory scopeFacto
         /// Received request/notification payload from client.
         /// </summary>
         public void OnDataReceived(string jsonRpcPayload)
+        {
+            OnDataReceived(jsonRpcPayload, null);
+        }
+
+        /// <summary>
+        /// Received request/notification payload from client.
+        /// </summary>
+        public void OnDataReceived(string jsonRpcPayload, ParserOptions options)
```

Comment:
>, ParserOptions options) [](http://example.com/codeflow?start=56&length=24)

We can't add a new parameter here because this is the existing LSP protocol. 
The formula bar is calling this, and it's only passing (string contents, cursor position). 

Extra information needs to come from the ScopeFactory and then get saved as properties in the scope interface.